### PR TITLE
Register MessagePack types with factories

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: integration/default_set.yaml [hunk 1]
```diff
@@ -152,11 +152,13 @@ tests:
     steps:
       - command: import -b --table-slice-type=caf zeek
         input: data/zeek/conn.log.gz
-      - command: import -b --table-slice-type=arrow zeek
-        input: data/zeek/conn.log.gz
       - command: import -b --table-slice-type=msgpack zeek
         input: data/zeek/conn.log.gz
       - command: count '#type == "zeek.conn"'
+      - guard: 'export -h 2>&1 | grep -q arrow'
```
Comment (mavam) [inline integration/default_set.yaml:158]: Could this be reformulated more idiomatically by using the `version` command?

Comment (dominiklohmann) [reply integration/default_set.yaml:158]: Yeah, like `version | jq -re '."Apache Arrow" != null'

Comment (dominiklohmann) [reply integration/default_set.yaml:158]: I've implemented this change.

Comment (dominiklohmann) [reply integration/default_set.yaml:158]: The push somehow doesn't show up in here, I think the GitHub UI is broken.

File: libvast/CMakeLists.txt [hunk 2]
```diff
@@ -244,7 +244,7 @@ if (VAST_USE_OPENSSL)
   target_link_libraries(libvast PUBLIC caf::openssl)
 endif ()
 
-target_link_libraries(libvast PRIVATE flatbuffers::flatbuffers)
+target_link_libraries(libvast PUBLIC flatbuffers::flatbuffers)
```
Comment (tobim) [inline libvast/CMakeLists.txt:247]: Is there an unavoidable reason that flatbuffers leak into the interface here?

Comment (dominiklohmann) [reply libvast/CMakeLists.txt:247]: Yeah, zeek-to-vast doesn't build without it. I'd use the INTERFACE target, but that's broken unless we bump the CMake version (a lot).

**Pull request merged**